core/src/main/java/org/axonframework/queryhandling/SimpleQueryBus.java

```diff
@@ -0,0 +1,123 @@
+package org.axonframework.queryhandling;
+
+import org.axonframework.common.Registration;
+import org.axonframework.messaging.MessageHandler;
+
+import java.text.MessageFormat;
+import java.util.Iterator;
+import java.util.Set;
+import java.util.Spliterator;
+import java.util.concurrent.*;
+import java.util.function.Consumer;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+/**
+ * Implementation of the QueryBus that dispatches queries to the handlers.
+ * @author Marc Gathier
+ * @since 3.1
+ */
+public class SimpleQueryBus implements QueryBus {
+    private final ConcurrentMap<QueryDefinition, Set<MessageHandler<? super QueryMessage<?>>>> subscriptions = new ConcurrentHashMap<>();
+
+    @Override
+    public Registration subscribe(String queryName, String responseName, MessageHandler<? super QueryMessage<?>> handler) {
+        QueryDefinition registrationKey = new QueryDefinition(queryName, responseName);
+        subscriptions.computeIfAbsent(registrationKey, (k) -> new CopyOnWriteArraySet<>()).add(handler);
+        return () -> unsubscribe(registrationKey, handler);
+    }
+
+    @Override
+    public <Q, R> CompletableFuture<R> query(QueryMessage<Q> query) {
+        CompletableFuture<R> completableFuture = new CompletableFuture<>();
+        Set<MessageHandler<? super QueryMessage<?>>> handlers = getHandlersForMessage(query);
+        try {
+            completableFuture.complete((R)handlers.iterator().next().handle(query));
+        } catch (Exception e) {
+            completableFuture.completeExceptionally(e);
+        }
+        return completableFuture;
+    }
+
+    private boolean unsubscribe(QueryDefinition registrationKey, MessageHandler<? super QueryMessage<?>> handler) {
+        subscriptions.computeIfPresent(registrationKey, (key, handlers) -> {
+            handlers.remove(handler);
+            if( handlers.isEmpty()) return null;
```

Comment:
I thought it's not desired to do `if/else` blocks without curly braces.
Additionally, indenting is a little of on the `if` statement.